Milestone 01

jonahtwalker opened the pull request:
<!-- This template is to serve as a comprehensive outline for documenting updates to the code. Some of these sections may not be relevant to your PR. In that case, document it inline with `N/A`  -->

## What does this PR do?
<!-- Provide a clear and concise description of what this PR accomplishes -->
This is the initial starting point for our Svelte app. It amounts to a portal with auth flows and basic user management features.

### Related Issues & PRs
<!-- List the related issues, using either: `Closes #123` or `Fixes #123` , which will automatically move the specified issue to "In Review" when the PR is created -->


#### Key Decisions
<!-- Explain the technical decisions and alternative approaches considered for your new implementations and refactored solutions -->
- **Implemeneted Approach**:
  <!-- What did you decide on? -->
  To keep it as simple as possible to begin, I did not opt into a testing library and stuck with `npm` as the package manager. These can be added after the fact, of course. There is documentation on all of the presets.
- **Alternatives Considered**:
  <!-- What else did you consider? -->
- **Rationale**:
  <!-- Why did you decide this was the most appropriate solution? -->
  Just for the sake of a tidy, simple start

## Breaking Changes
<!-- Do others need to do something to successfully incorporate your changes into their work? -->
- [ ] This PR includes breaking changes
  <!-- What do they need to do? -->
Make sure you `pnpm i`! 

### Limitations & Concerns
<!-- Describe any limitations, potential edge cases, or situations where your solution might break -->
- **Known limitations**: This is just a scaffold.
- **Edge cases**:
- **Potential risks**:

---

### UI Changes
<!-- If this PR modifies the UI, provide screenshots and descriptions -->
- [x] This PR includes UI changes
<!-- Include screenshots in the table below, include before & after applicable -->

### Screenshots


---

### Reviewers
<!-- Tag at least 2 others for review -->
- [x] Reviewer 1: @RaghvindYadav
- [ ] Reviewer 2: @rupalirajput2597

### Next Steps
<!-- What needs to happen after this PR is merged? -->
- [ ] Follow-up tasks are created
  - [ ] Decide on test framework & implement it
  - [ ] Upgrade from `npm` to the modern `yarn` or something else, like `pnpm`

### How to Test
<!-- Provide detailed steps for QA to test the changes -->
1. Prerequisites:
   <!-- e.g. Do I need to be a specific role? -->
2. Test steps:
   <!-- What is the quickest path to testing your solution? -->
3. Expected results:
   <!-- Is there any nuance that should be clarified? If so, can the implementation get more simple instead? -->
   The few components there should all be interactive - but they all contain placeholder actions or routes.

## Final Checklist
- [ ] Linked related issues
- [x] Rebased with target branch
- [x] No lint errors
- [x] QA'd in Chrome, Safari & Firefox
- [x] Added/updated tests (if applicable)
- [x] Mobile responsive (if applicable)
- [x] Updated documentation (if applicable)
- [x] Notified the Team of this updates readiness

---

## AI Usage
<!-- Document any AI tools you used to develop this -->
- [x] AI tools were used in development
  - **Tool used**:
    <!-- Specify any model used -->
  - **Prompts used**:
    <!-- Format these inclusions as quotes, like: `> Provide  -->
    - **Initial prompt**: 
- [] No AI tools were used
  - **Reason**:
    <!-- Why not? -->

## References
<!-- List any helpful references used during development -->
- **Documentation**:
 - [SvelteKit Getting Started](https://svelte.dev/docs/kit/creating-a-project)
 - [shadcn-svelte v5 docs](https://next.shadcn-svelte.com/)
 - [Tailwind core concepts](https://v3.tailwindcss.com/docs/responsive-design)
- **Discussions**:
- **Related PRs**:

jonahtwalker committed:
Rebased onto main

Commit:
Removed IDEA files

jonahtwalker committed:
tweak pull_request_template.md (#2)

RaghvindYadav reviewed (dismissed):
Code looks good, we could merge it. I added a commit which can be tackle in next PR.

Commit:
landing-page

Commit:
mobile view

jahanvi-baragaun committed:
moved sidebarbutton in sidebar when it is open (#24)

Co-authored-by: Jahanvi <jahanvi@Jahanvis-MacBook-Air.local>

Commit:
commit fixes

RaghvindYadav committed:
Merge pull request #26 from baragaun/ry/landing-page

The Landing Page

Commit:
authentication

Commit:
Update +page.svelte

Commit:
comment fix

Commit:
recovery code

evanschimpf committed:
Switched to pnpm, updated README (#31)

Co-authored-by: evan <evan@baragaun.com>

jonahtwalker commented:
@RaghvindYadav @rupalirajput2597 @jahanvi-baragaun 

Evan provided a PR to switch from `npm` to `pnpm`, please check out his update that contains install instructions. We will need to rebase the feature branches. 👍🏼 

#31

Commit:
authentication

Commit:
Added .nvmrc, README wording

Commit:
Modified prettier config, applied it

Commit:
formatting

picked the #33

Commit:
Documentation

Commit:
Added prettier-plugin-organize-imports and ran format

Commit:
corrections

Commit:
wording

Commit:
Merge remote-tracking branch 'origin/ry/new-authentication' into ry/new-authentication

Commit:
Update form-libraries.md

Commit:
doc format

RaghvindYadav committed:
Delete .repomix/bundles.json

RaghvindYadav committed:
Merge pull request #32 from baragaun/ry/new-authentication

Authentication: SignIn, Create Account and Forgot Password

Commit:
Added bg-node-client

Commit:
Added bg-node-client

Commit:
bg-node-client dep link fix

Commit:
pnpm approve-builds update for rxdb, protobufjs, libsignal, firebase

Commit:
Adapted changes to fsdata

Commit:
fsdata changes

Commit:
fsdata changes

rupalirajput2597 committed:
feature/footer-section: implemented privacy-policy, faq, and ToS (#25)

* added footer-section
* implemented privacy-policy, ToS, and FAQ with dummy data
* added dev-env.txt file
* Updated README

jahanvi-baragaun committed:
playwright added, landing screen test added, Vitest added and unit test added for  landing screen. (#29)

* playwright added, landing screen test added
* vitest added and sample unit test added
* gitignore file changes
* steps added in README.md file to run different tests
* Svelte5 LLM Medium file added from svelte site
* adds github action to run all tests

Co-authored-by: Jonah <64275677+jonahtwalker@users.noreply.github.com>

Commit:
updated testing yml to use .nvmrc for node version

evanschimpf committed:
Merge pull request #37 from baragaun/es/testing-node-version-fix

Use node version from .nvmrc in testing workflow

Commit:
changed to single quotes for consistency

Commit:
fsdata changes

Commit:
Onboarding flow fixes

Commit:
Text changes

Commit:
yarn run format

Commit:
eslint fix - ignore unused variables

Commit:
eslint no-unused-vars disabled globally

rupalirajput2597 committed:
Rr/milestone 01 account setting (#34)

* Feature: Setting page
* includes Account Settings, Notification Settings
* added notifications-setting option along with account setting
* added dialog to edit account setting

jonahtwalker commented:
@shadowbrush @evanschimpf Can we PLEASE document what happens in these PRs? All of the documentation is in the other issues and other PRs, but then things just appear here.

Why integrate the node-client right into Milestone 1 when there is an issue and a dedicated branch for it? Why is this branch any different than any other feature or development branch?

It's a rebase fest every single day, all the while we are asking the team to compartmentalize and componentize their contributions. I don't want to juggle development branches if we're just going to test in our version of Production today anyways.

jonahtwalker committed:
Scaffold Refactors (#44)

- added bg-node-client & rxdb
- added zod
- added MyUserContext
- added some documentation on the data layer
- initial unit test setup
- initial playwright setup
- initial storybook set up
- many component refactors
- many page and template refactors

Co-authored-by: Jonah <64275677+jonahtwalker@users.noreply.github.com>
Co-authored-by: rupalirajput2597 <46842523+rupalirajput2597@users.noreply.github.com>
Co-authored-by: Raghvind Yadav <123397631+RaghvindYadav@users.noreply.github.com>
Co-authored-by: shadowbrush <github@shadowbrush.com>

jonahtwalker committed:
fix vitest setup global error

shadowbrush commented on src/lib/contexts/my-user-context.svelte.ts:
```diff
@@ -0,0 +1,704 @@
+  private polling: QueryPollingOptions = {
+    enabled: true,
+    interval: 1000,
+    timeout: 15 * 10 * 1000,
```
@RaghvindYadav `15 * 10 * 1000` are 2.5 minutes, not 1.5. 

For a `MultiStepAction` we may want to increase this to 15 minutes, with a 2 second interval. Remember that people can get distracted in the middle of onboarding, i.e. with a phone call, or a crying child.

Polling is also used when updating an object, i.e. `MyUser`. For those, we can use 30 seconds timeout and 1 second interval.

shadowbrush commented on src/lib/contexts/my-user-context.svelte.ts:
```diff
@@ -0,0 +1,704 @@
+    // Timeout should parallel to token expiry time, for now it is 1.5 mins it enought to user to verify and send another token.
+  };
+  // Add these methods for event handling
+  public addEventListener(eventType: string, callback: (event: UserContextEvent) => void): void {
```
@RaghvindYadav `addEventListener` and `removeEventListener` are never used, or are they? What are they needed for? The whole subscribing to an event from BgNodeClient and publishing it again is a bad design. Any component can subscribe to BgNodeClient's events, so there is no need for the context to relay them. We need to remove this.

shadowbrush commented on src/lib/contexts/my-user-context.svelte.ts:
```diff
@@ -0,0 +1,704 @@
+    // After successful initialization, check if user is signed in and restore user data
+    if (this.client.isSignedIn) {
+      try {
+        const userResponse = await this.client.operations.myUser.findMyUser();
```
@RaghvindYadav As I mentioned on Slack, there is no need to call `findMyUser`. The MyUser object will be published to this context automatically. You can check `this.client.isSignedIn`, but calling `findMyUser` serves no purpose. Delete.

RaghvindYadav replied:
I know you mentioned that in the slack. I called, it intentionally when you did reload or refresh the page then user data gets cleared. To test this you just need to follow avatar component.

shadowbrush commented on src/routes/signup/form/sign-up-form.svelte:
```diff
@@ -0,0 +1,217 @@
+    try {
+      console.log('Email submitted:', email);
+
+      const response = await myUserContext.verifyMyEmail(email);
```
@RaghvindYadav `response` is of type `string | true`. If you want to check whether there's an error, do this:

```
if (response !== true)
```

This means line 113 will always think there is an error, even if `response` is `true`.

RaghvindYadav replied:
okay noted.

shadowbrush commented on src/routes/signup/form/sign-up-form.svelte:
```diff
@@ -0,0 +1,217 @@
+        return;
+      }
+      console.log('Email confirmation started:', response);
+      currentStep.set(1);
```
@RaghvindYadav Don't go to step `1` until you know whether the email went out or failed. If the email failed to send, this page needs to show an error message, and not go to the next step. The user should provide an email that allows us to send the email successfully.

This weekend I added a listener to this component in this function. You deleted it. Put it back in. The component needs to know what's going on with the multi step action.

RaghvindYadav replied:
Yes I deleted it because you mentioned that you will change the Response output in `true | string` so I tried to make  a similar approach every where. I added listener here too just like I added in signIn- form.

RaghvindYadav replied:
> the email went out or failed.

This happening all time so I think we need to first sort out node-mailer issues.
I will ignore it only for development. you know we are using random emails for signUp.

shadowbrush replied:
It's now ignored for the `development` environment.

Commit:
Onboarding flow

shadowbrush committed:
Added lint rules (#61)

* Not allowing nested ternaries
* Forcing single quotes
* Forcing semicolons
* Forcing trailing commas

Co-authored-by: shadowbrush <github@shadowbrush.com>

evanschimpf committed:
Es/unit test fix (#70)

* fixed unit tests
* CI update
* README and package.json update

---------

Co-authored-by: evan <evan@baragaun.com>

Commit:
Fixed e2e test

Commit:
cleanup

evanschimpf committed:
Merge pull request #73 from baragaun/es/e2e-test-fix

Es/e2e test fix

evanschimpf committed:
Es/paraglide js - Paraglide JS Language Localization (#64)

* Installed Paraglide JS
* paraglide js implemented
* updated existing tests

---------

Co-authored-by: evan <evan@baragaun.com>
Co-authored-by: shadowbrush <github@shadowbrush.com>
Co-authored-by: Jonah <64275677+jonahtwalker@users.noreply.github.com>

RaghvindYadav committed:
Reset Password (#59)

* Refactor forms to use Formsnap & Zod
* Adds Reset Password flow

---------

Co-authored-by: shadowbrush <github@shadowbrush.com>
Co-authored-by: Jonah <64275677+jonahtwalker@users.noreply.github.com>
Co-authored-by: Rupali Rajput <rupali@Mac-7.lan>

Commit:
Added instructions for paraglide

Commit:
Added file to track translation TODOs

Commit:
documentation update

evanschimpf committed:
Merge pull request #80 from baragaun/es/paraglide-js-instructions

Es/paraglide js instructions

Commit:
lint & prettier fixes

Commit:
prettier format fix

Commit:
vite upgrade (CVEs)

Commit:
Added llm context & instructions

jonahtwalker committed:
Refactor ResetPassword flow to be two steps (#88)

* Refactored ResetPassword flow to be 2 steps

jonahtwalker committed:
Refactor ResetPassword flow to be two steps (#88)

* Refactored ResetPassword flow to be 2 steps

Commit:
pnpm install & approve-builds

evanschimpf committed:
Merge branch 'milestone-01' into es/eslint-prettier-update

evanschimpf committed:
Merge pull request #87 from baragaun/es/eslint-prettier-update

Es/eslint prettier update

Commit:
lint/prettier formatting fixes

Commit:
Upgraded version of BgNodeClient to 0.0.27

Commit:
Upgraded version of BgNodeClient to 0.0.28

rupalirajput2597 committed:
added fix for sidebar auto collapes on small devices (#93)

* added fix for sidebar auto collapes on small devices
* updated @baragaun/bg-node-client

rupalirajput2597 committed:
RR/account settings with superform and form components (#85)

Refactors settings/account forms
---------
Co-authored-by: Jonah <64275677+jonahtwalker@users.noreply.github.com>

rupalirajput2597 committed:
Update the current ThemeButton : Issue #79 (#92)

Refactored ThemeButton to become LightSwitch
- Refactored avatarmenu and navbar

---------
Co-authored-by: Jonah <64275677+jonahtwalker@users.noreply.github.com>

RaghvindYadav committed:
Added a Connection indicator component to the Sidebar

- Added connectivity Sonners

---------

Co-authored-by: Jonah <64275677+jonahtwalker@users.noreply.github.com>

rupalirajput2597 committed:
Rr/unauthenticated UI update (#97)

Updates what is visible when anonymous
- Misc. ui fixes
- Sidebar CTA addition

rupalirajput2597 committed:
RR/bugfix and translation updated (#120)

Fixes: #102, #110, #111, #112, #113, #115, #117, #119

---------

Co-authored-by: Rupali Rajput <rupali@Mac.lan>

jonahtwalker committed:
Refactor myUserContext out of base components (#127)

* Refactored myUserContext out of base components

rupalirajput2597 committed:
signin-signup and reset password screen translation and bugfixing (#123)

* Bug fixes and signup flow translations
---------
Co-authored-by: Jonah <64275677+jonahtwalker@users.noreply.github.com>

jonahtwalker committed:
Translations and global layout updates (#132)

- add logic to catch chromium svelekit error
- disable and redirect from the settings/notifications
- update translations
- add tooltips to nav icon buttons
- rename `components/global` to `components/layout`
- relocated language and light switch buttons

RaghvindYadav committed:
Story of components (#131)

Fixes the Vitest & Storybook config
- adds most component stories
- some stories missing controls / functionally incomplete

---------

Co-authored-by: Jonah <64275677+jonahtwalker@users.noreply.github.com>

rupalirajput2597 committed:
RR/Fix small screen dialog UI and duplicate email validation (#140)

* Resolved issues: #139 and #138
---------
Co-authored-by: Jonah <64275677+jonahtwalker@users.noreply.github.com>

jonahtwalker committed:
Handle onboarding progress (#144)

* Fixes #125, #126 & #147

---------

Co-authored-by: rupalirajput2597 <46842523+rupalirajput2597@users.noreply.github.com>

evanschimpf committed:
Added turnstile (#143)

Co-authored-by: Jonah <64275677+jonahtwalker@users.noreply.github.com>

evanschimpf committed:
Updated example .env for turnstile

rupalirajput2597 committed:
RR/bugfixes: resolved issue #148 #150 #151 (#154)

* Bugfixes : Resolves issue #148 #150 #151
* updates: bg-node-client version

rupalirajput2597 committed:
bugfix-token-error-not-showing-again: #146 #149 (#155)

evanschimpf committed:
Example .env update

evanschimpf committed:
.env variable refactor

evanschimpf committed:
prettier format

evanschimpf committed:
Merge pull request #162 from baragaun/es/env-refactor

Es/env refactor

jonahtwalker committed:
Initial form state & direct context use (#158)

* Update initial form states
* Refactor direct use of MyUserContext

rupalirajput2597 committed:
#160: fix-Update email: No error message shown after first incorrect … (#161)

* #160: fix-Update email: No error message shown after first incorrect token attempt
* updated lockfile

---------

Co-authored-by: Jonah <64275677+jonahtwalker@users.noreply.github.com>

shadowbrush committed:
Upgraded bg-node-client to 0.0.42 (#163)

Co-authored-by: shadowbrush <github@shadowbrush.com>
